content/en/docs/zero-code/java/agent/extensions.md

```diff
@@ -49,8 +68,1083 @@ extensions:
   existing `Tracer` directly or extend it. As you have your own `Advice`, you
   can control which `Tracer` you use.
 
-## Extension examples
+## Loading an Extension at Runtime
+
+There are two ways to use extensions with the Java agent:
+
+1. **Load at runtime** using the `-Dotel.javaagent.extensions` option (described
+   in this section)
+2. **Embed in the agent** to create a single JAR file (described in the
+   [next section](#embedding-extensions-in-the-agent))
+
+### Basic Usage
+
+Extensions can be loaded at runtime using the `otel.javaagent.extensions` system
+property. This property accepts comma-separated paths to extension JAR files or
+directories containing extension JARs.
+
+#### Single Extension
+
+```bash
+java -javaagent:path/to/opentelemetry-javaagent.jar \
+     -Dotel.javaagent.extensions=/path/to/my-extension.jar \
+     -jar myapp.jar
+```
+
+#### Multiple Extensions
+
+```bash
+java -javaagent:path/to/opentelemetry-javaagent.jar \
+     -Dotel.javaagent.extensions=/path/to/extension1.jar,/path/to/extension2.jar \
+     -jar myapp.jar
+```
+
+#### Extension Directory
+
+You can specify a directory containing multiple extension JARs, and all JARs in
+that directory will be loaded:
+
+```bash
+java -javaagent:path/to/opentelemetry-javaagent.jar \
+     -Dotel.javaagent.extensions=/path/to/extensions-directory \
+     -jar myapp.jar
+```
+
+#### Mixed Paths
+
+You can combine individual JAR files and directories:
+
+```bash
+java -javaagent:path/to/opentelemetry-javaagent.jar \
+     -Dotel.javaagent.extensions=/path/to/extension1.jar,/opt/extensions,/tmp/custom.jar \
+     -jar myapp.jar
+```
+
+### How Extension Loading Works
+
+When you load extensions at runtime, the agent:
+
+1. Creates an
+   [isolated class loader](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/javaagent-structure.md#extension-class-loader)
+   for each extension JAR to prevent conflicts between extensions
+2. Makes OpenTelemetry APIs available to your extension (without needing to
+   package them in your extension JAR)
+3. Automatically applies shading to extension classes at runtime (extensions use
+   normal OpenTelemetry imports, and the agent handles compatibility
+   automatically)
```

Comment:
I think the shading is kind of an advanced topic, hopefully most of the extension users don't even realize that something is shaded. If we are going to mention it then I think the goal should be to reduce confusion for the ones who try to debug the extension and notice that things aren't quite as they'd expect.